File: Python/283_Move_Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """

        count = 0
        for num in nums:
            if num != 0:
                nums[count] = num
                count += 1

        for i in range(count, len(nums)):
            nums[i] = 0

        return nums

# algorithm: count number not zero 
# time O(N)
# memory O(N)

File: C++/283_MoveZeroes.cpp
using namespace std;
#include <vector>

class Solution {
public:
    void moveZeroes(vector<int>& nums) {

        int count = 0;

        for (auto num:nums){
            if (num != 0) {
                nums[count] = num;
                count++;
            }
        }

        for (int i=count;i<nums.size();i++){
            nums[i] = 0;
        }        
    }
};
